packages/react-instantsearch/src/core/utils.js

```diff
@@ -58,3 +58,6 @@ export function assertFacetDefined(searchParameters, searchResults, facet) {
 export function getDisplayName(Component) {
   return Component.displayName || Component.name || 'UnknownComponent';
 }
+
+const resolved = typeof Promise !== 'undefined' && Promise.resolve();
+export const defer = resolved ? f => { resolved.then(f); } : setTimeout;
```

Comment:
After some searching it seems the current fastest way to express "Please execute this code asynchronously as soon as possible" is to actually use Promise.resolve().then().

https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules and https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/#comment-2926206647

I also asked @developit for his advice on this, we can wait a bit before merging.